miner/worker.go

```diff
@@ -834,14 +854,25 @@ func (miner *Miner) validateParams(genParams *generateParams) (time.Duration, er
 	return time.Duration(blockTime) * time.Second, nil
 }
 
-func (miner *Miner) blockFeeCurrency(env *environment, feeCurrency common.Address, err error) {
-	// the fee-currency is still in the allowlist of this environment,
-	// so set the fee-currency block gas limit to 0 to prevent other
-	// transactions.
-	pool := env.multiGasPool.PoolFor(&feeCurrency)
-	pool.SetGas(0)
-	// also add the fee-currency to a worker-wide blocklist,
-	// so that they are not allowlisted in the following blocks
-	// (only locally in the txpool, not consensus-critical)
-	miner.feeCurrencyBlocklist.Add(feeCurrency, *env.header)
+func (miner *Miner) registerFeeCurrencyTxFailure(env *environment, tx *types.Transaction, err error) {
+	if !env.blocklistsDisabled {
+		// the fee-currency is still in the allowlist of this environment,
+		// so set the fee-currency block gas limit to 0 to prevent other
+		// transactions.
+		pool, hasSeparateMultiPool := env.multiGasPool.PoolFor(tx.FeeCurrency())
+		// if for whatever reason we didn't set a separate multipool
+		// for this fee-currency, we don't want to completely
+		// block all other gas usage
```

Comment:
Ahh, well spotted!

Reply:
This might warrant a log message though, since this case is not expected